containers/init/ready/ready.go

```diff
@@ -252,6 +257,50 @@ func WaitForReadyPods(ctx context.Context, ltg LoadTestGetter, pl PodLister, tes
 	return podAddresses, &nodesInfo, nil
 }
 
+func passTarget(quitServer bool, clientIP string, targets []*pb.Endpoint, testType string) error {
+	xdsEndpointUpdatePort := os.Getenv(testconfig.XDSEndpointUpdatePortEnv)
+	if xdsEndpointUpdatePort == "" {
+		log.Fatalf("failed to obtain the xds endpoint update server port for PSM test")
+	}
+	dialTarget := net.JoinHostPort(clientIP, xdsEndpointUpdatePort)
+	conn, err := grpc.Dial(dialTarget, grpc.WithInsecure())
+	if err != nil {
+		log.Fatalf("did not connect: %v", err)
+	}
+	defer conn.Close()
+	c := pb.NewTestUpdaterClient(conn)
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+	if _, err := c.UpdateTest(ctx, &pb.TestUpdateRequest{Endpoints: targets, TestType: testType}); err != nil {
+		log.Fatalf("could not greet: %v", err)
+		statusCode, _ := grpcstatus.FromError(err)
+		log.Print(statusCode.Details()...)
+	}
+	// Used separate goroutines to stop server, non-blocking
+	if quitServer {
+		go func() {
+			fmt.Println("stopping endpoint update server")
+			if _, err := c.QuitTestUpdateServer(ctx, &pb.Void{}); err != nil {
+				statusCode, _ := grpcstatus.FromError(err)
```

Comment:
How critical is it if this RPC fails?

Reply:
My understanding is since it was separated to a different go-routine, it would not stop the ready container from moving forward. The endpoint (I should have updated the name, I will fix it) update server in the xds server is also in a separate go-routine, I don't think the failure would do anything. Not able to quit it may just cause it to be continuously running.

Reply:
Closing this thread in favor of [this one](https://github.com/grpc/test-infra/pull/255#discussion_r798138526) since they're essentially the same question.